<a href="https://colab.research.google.com/github/pranilthorat/capstone-project-almabetter/blob/main/github_airbnb_capstone_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

* **AIRBNB** is an American company that operates an online marketplace for lodging, primarily homestays for vacation rentals, and tourism activities. Based in San Francisco, California, the platform is accessible via website and mobile app. Airbnb does not own any of the listed properties; instead, it profits by receiving commission from each booking. The company was founded in 2008 by Brian Chesky, Nathan Blecharczyk and Joe Gebbia. Airbnb is a shortened version of its original name, AirBedandBreakfast.com.

In [18]:
#import pandas,numpy,seaborn,matplotlib,warnings
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings(action='ignore')

Data Frame of Airbnb

In [19]:
#mount drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [20]:
#dataframe of airbnb_df
airbnb="/content/drive/MyDrive/Airbnb NYC 2019.csv"
airbnb_df=pd.read_csv(airbnb)


In [21]:
#head of airbnb_df
airbnb_df.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

In [22]:
#tail of airbnb_df
airbnb_df.tail()

id  ... availability_365
48890  36484665  ...                9
48891  36485057  ...               36
48892  36485431  ...               27
48893  36485609  ...                2
48894  36487245  ...               23

[5 rows x 16 columns]

In [23]:
#checking the size of the airbnb_df
airbnb_df.size

782320

In [24]:
#checking shape of airbnb_df
airbnb_df.shape

(48895, 16)


cleaning the data▶

In [25]:
#checking null values or unknown data 
airbnb_df.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [26]:
#getting thoes colunms which is required for analysis of hosts 
airbnb_df_new=airbnb_df.loc[:,['host_id','host_name','price','number_of_reviews','reviews_per_month','availability_365']]
airbnb_df_new   

host_id      host_name  ...  reviews_per_month  availability_365
0          2787           John  ...               0.21               365
1          2845       Jennifer  ...               0.38               355
2          4632      Elisabeth  ...                NaN               365
3          4869    LisaRoxanne  ...               4.64               194
4          7192          Laura  ...               0.10                 0
...         ...            ...  ...                ...               ...
48890   8232441        Sabrina  ...                NaN                 9
48891   6570630        Marisol  ...                NaN                36
48892  23492952  Ilgar & Aysel  ...                NaN                27
48893  30985759            Taz  ...                NaN                 2
48894  68119814     Christophe  ...                NaN                23

[48895 rows x 6 columns]

In [27]:
#handeling missing values 
airbnb_df_new.fillna({'reviews_per_month':0}, inplace=True)
airbnb_df_new.dropna(subset=['host_name'],inplace=True)

In [28]:
#lenght of new airbnb_df
len(airbnb_df_new)

48874

In [29]:
#checking null values or missing data 
airbnb_df_new.isnull().sum()

host_id              0
host_name            0
price                0
number_of_reviews    0
reviews_per_month    0
availability_365     0
dtype: int64

In [30]:
#we have clean data now
print(type(airbnb_df_new["host_id"]))
print(type(airbnb_df_new['host_name']))
print(type(airbnb_df_new["availability_365"]))
print(type(airbnb_df_new["number_of_reviews"]))
print(type(airbnb_df_new["reviews_per_month"]))
print(type(airbnb_df_new["price"]))

<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>


In [31]:
#head of airbnb_df_new
airbnb_df_new.head()

host_id    host_name  ...  reviews_per_month  availability_365
0     2787         John  ...               0.21               365
1     2845     Jennifer  ...               0.38               355
2     4632    Elisabeth  ...               0.00               365
3     4869  LisaRoxanne  ...               4.64               194
4     7192        Laura  ...               0.10                 0

[5 rows x 6 columns]

In [32]:
#tail of airbnb_df_new
airbnb_df_new.tail()

host_id      host_name  ...  reviews_per_month  availability_365
48890   8232441        Sabrina  ...                0.0                 9
48891   6570630        Marisol  ...                0.0                36
48892  23492952  Ilgar & Aysel  ...                0.0                27
48893  30985759            Taz  ...                0.0                 2
48894  68119814     Christophe  ...                0.0                23

[5 rows x 6 columns]

# Now, we have to Explore and analyze the data to discover key understandings such as : 
 * What can we learn about different hosts and areas?
Let us first start with Hosts,
Well we can learn following points about hosts

# 1] There are how many hosts which has availability of 365 days i.e 24x7 
# 2] which host has the highest	number of reviews
# 3] which host has the highest number of reviews per month 
# 4] which host has most expensive room per night
# 5] which host has cheapest room per night

1] There are how many hosts which has availability of 365 days i.e 24x7

In [33]:
#lets see max days availability host has 
airbnb_df_new["availability_365"].max()

365

In [34]:
#how many hosts has availability of 365 days
availability_365_days=airbnb_df_new.loc[:,['host_id','availability_365']]
host_available_365_days=availability_365_days[availability_365_days['availability_365']>364]
how_many_host_available_for_365_days=host_available_365_days['host_id'].nunique()
print(how_many_host_available_for_365_days)

894


so there are total 894 hosts which has availability of 365 days